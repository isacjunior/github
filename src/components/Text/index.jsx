import React from 'react'
import PropTypes from 'prop-types'
import TextStyled from './styled'

const Text = ({ children, bold, cursor }) => (
  <TextStyled bold={bold} cursor={cursor}>
    {children}
  </TextStyled>
)

Text.defaultProps = {
  bold: false,
  cursor: 'normal'
}

Text.propTypes = {
  children: PropTypes.node.isRequired,
  bold: PropTypes.bool,
  cursor: PropTypes.string
}

export default Text
