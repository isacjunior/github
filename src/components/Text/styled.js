import styled from 'styled-components'

const TextStyled = styled.span`
  color: #556271;
  font-size: 16px;
  font-weight: ${props => props.bold ? 'bold' : 'normal'};
`

export default TextStyled
