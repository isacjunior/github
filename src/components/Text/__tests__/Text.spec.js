import React from 'react'
import { mount } from 'enzyme'
import Text from '../index'

describe('Text', () => {
  it('Text render', () => {
    mount(<Text>Children</Text>)
  })

  it('Snapshot Text', () => {
    const wrapper = mount(<Text>Children</Text>)
    expect(wrapper).toMatchSnapshot()
  })
})
