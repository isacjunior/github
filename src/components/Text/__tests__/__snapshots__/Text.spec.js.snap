// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Snapshot Text 1`] = `
<Text
  bold={false}
>
  <styled.span
    bold={false}
  >
    <span
      className="sc-bdVaJa cPuCgi"
    >
      Children
    </span>
  </styled.span>
</Text>
`;
